src/vm/vm_core.rs

```diff
@@ -3829,4 +3835,24 @@ mod tests {
             ]),
         );
     }
+
+    /// Test that compute_effective_sizes() works as intended.
+    #[test]
+    fn compute_effective_sizes() {
+        let mut vm = vm!();
+
+        let segment = vm.segments.add(&mut vm.memory);
+        vm.load_data(
+            &segment.into(),
+            vec![
+                mayberelocatable!(1),
+                mayberelocatable!(2),
+                mayberelocatable!(3),
+                mayberelocatable!(4),
+            ],
+        )
+        .expect("Could not load data into memory.");
```

Comment:
Let's use `.unwrap()` here, in order in order to show the real panic error.

Reply:
I'm not sure an `.unwrap()` would be better in this case. [Here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fc8f63a2869da8eff7bcf720c77c8d7d)'s an example.

The difference is bettween getting
  - `thread 'main' panicked at 'called 'Result::unwrap()' on an 'Err' value: TestError(1234)', src/main.rs:12:50` with an unwrap, or
  - `thread 'main' panicked at 'An error occurred: TestError(1234)', src/main.rs:13:50` with an expect.